### ISSUE TITLE
Switch to the OpenAPI 3.1 definitions

### ISSUE THREAD

**Issue by wolfy1339** [octokit/openapi#344]
### Describe the need

GitHub started publishing OpenAPI 3.1 schemas, which now include definitions for webhooks.

It would be nice to have them published in this repo.
See also, https://github.com/octokit/webhooks/issues/470

This would be blocked by #321

In the mean time, we could have a separate repo which contains only the webhook spec, until we can find a replacement
I have created a PoC in my repo: https://github.com/wolfy1339/openapi-webhooks and I will be more than happy to transfer it over to the Octokit org

### SDK Version

N/A

### API Version

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
